src/data_cleaning.py

```diff
@@ -716,6 +721,15 @@ def calculate_aggregated_primary_fuel(
         f"{level}_primary_fuel"
     ].fillna(agg_primary_fuel["primary_fuel_from_mode"])
 
+    # sometimes nuclear generators report 0 fuel consumption in EIA-923.
+    # To ensure that a nuclear plant does not get assigned a fuel code of
+    # a backup generator, we use the nameplate capacity to assign the
+    # primary fuel for any plants that contain a nuclear generator
+    agg_primary_fuel.loc[
+        agg_primary_fuel["primary_fuel_from_capacity_mw"] == "NUC",
+        f"{level}_primary_fuel",
+    ] = "NUC"
```

Comment:
Is this also the case for any other generators that don't use a fossil fuel as their primary? (ie, hydro, wind, solar)

Reply:
Good question. I did look at this for wind and hydro and didn't see any issues - they all report fuel consumption data. 

There are a number of entries in the EIA-923 generation and fuel table that report positive net generation but zero fuel consumption. Many of these are for `CA` prime movers, which means the steam portion of a combined cycle plant, which makes sense because the fuel is consumed at the `CT` prime mover. Ignoring those, the majority of the remaining data is nuclear steam turbines. There are a few other edge cases that wouldn't be caught by this step, but it's small enough that I don't think it is worth our time to address it now. I will however create an issue to investigate this in the future.